src/main/java/com/example/controllers/DashboardController.java

```diff
@@ -0,0 +1,206 @@
+package com.example.controllers;
+
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.FileNotFoundException;
+import java.io.FileOutputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.ObjectInputStream;
+import java.io.ObjectOutputStream;
+import java.io.OutputStream;
+import java.security.Principal;
+import java.util.List;
+
+import javax.servlet.http.HttpServletResponse;
+
+import com.example.bean.Account;
+import com.example.bean.CashAccount;
+import com.example.bean.CreditAccount;
+import com.example.bean.FileUntrusted;
+import com.example.bean.FileUntrustedValid;
+import com.example.dao.AccountDao;
+import com.example.dao.CashAccountDao;
+import com.example.dao.CreditAccountDao;
+import com.example.facade.StorageFacade;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Controller;
+import org.springframework.ui.Model;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestMethod;
+import org.springframework.web.bind.annotation.RequestParam;
+import org.springframework.web.bind.annotation.ResponseBody;
+import org.springframework.web.multipart.MultipartFile;
+import org.springframework.web.servlet.mvc.support.RedirectAttributes;
+
+@Controller
+@RequestMapping(value = "/dashboard")
+public class DashboardController {
+
+	@Autowired
+	AccountDao accountDao;
+
+	@Autowired
+	CashAccountDao cashaccountDao;
+
+	@Autowired
+	CreditAccountDao creditaccountDao;
+
+	@Autowired
+	StorageFacade storageFacade;
+
+	@RequestMapping
+	public String activity(final Model model, final Principal principal) {
+
+		Account account = accountDao.findUsersByUsername(principal.getName()).get(0);
+
+		List<CashAccount> cashAccounts = cashaccountDao.findCashAccountsByUsername(principal.getName());
+		List<CreditAccount> creditAccounts = creditaccountDao.findCreditAccountsByUsername(principal.getName());
+
+		model.addAttribute("account", account);
+		model.addAttribute("cashAccounts", cashAccounts);
+		model.addAttribute("creditAccounts", creditAccounts);
+		return "dashboard";
+	}
+
+	@RequestMapping(value = "/userDetail", method = RequestMethod.GET)
+	public String userDetail(final Model model, final Principal principal, @RequestParam(value = "username") final String username) {
+
+		Account account = accountDao.findUsersByUsername(username).get(0);
```

Comment:
#### Automated PR Comment From Polaris SAST
<!-- 4a7c1abeb16f787e48dde809216659dd -->
<!-- Comment managed by Git Commenter, do not modify! -->
### Polaris SAST Issue - SQL Injection
_High CWE-89_
Untrusted user-supplied data is inserted into a SQL statement without adequate validation, escaping, or filtering.

A user can change the intent of the SQL query, which may inappropriately disclose or corrupt data within the database.

### How to fix
Rewrite all SQL queries constructed through dynamic concatenation to use an injection-safe query mechanism such as prepared statements with parameterized queries.

Most modern programming languages provide a feature called "parameterized queries" that allow user-supplied data to be inserted safely as values in dynamic SQL queries. Rather than construct the dynamic SQL query by concatenating user-supplied data to static SQL query string fragments, data values are identified in the query by parameter markers or variables. Dynamic data is then passed through a mechanism provided by SQL that prevents the supplied data from changing the meaning of the query.

Note: the exact syntax and use of prepared statements with parameterized queries vary from language to language.